pkg/api/handlers/authprovider.go

```diff
@@ -221,6 +230,42 @@ func (ap *AuthProviderHandler) Deconfigure(req api.Context) error {
 		delete(ref.Annotations, v1.AuthProviderSyncAnnotation)
 	}
 
+	// Drop the sessions table and session_locks table from the database, if it exists.
+	if ap.postgresDSN != "" {
+		db, err := gorm.Open(postgres.Open(ap.postgresDSN), &gorm.Config{
+			Logger: logger.Default.LogMode(logger.Silent),
+		})
+		if err != nil {
+			return fmt.Errorf("failed to connect to postgres: %w", err)
+		}
+		sqlDB, err := db.DB()
+		if err != nil {
+			return fmt.Errorf("failed to get underlying sql.DB: %w", err)
+		}
+		defer sqlDB.Close()
+
+		var tablePrefix string
+		switch ref.Name {
+		case "github-auth-provider":
+			tablePrefix = "github_"
+		case "google-auth-provider":
+			tablePrefix = "google_"
+		case "entra-auth-provider":
+			tablePrefix = "entra_"
+		case "okta-auth-provider":
+			tablePrefix = "okta_"
+		}
+
+		if tablePrefix != "" {
+			if err := db.Exec("DROP TABLE IF EXISTS " + tablePrefix + "sessions;").Error; err != nil {
+				return fmt.Errorf("failed to drop sessions table: %w", err)
+			}
+			if err := db.Exec("DROP TABLE IF EXISTS " + tablePrefix + "session_locks;").Error; err != nil {
+				return fmt.Errorf("failed to drop session_locks table: %w", err)
+			}
+		}
+	}
+
```

Comment:
Feels like this is low-level implementation details bleeding up. Is it possible to move this into the auth providers themselves? maybe have them expose a `destroy` tool that does this instead?

If not, we should at least make the table prefix an field on the provider meta so we don't need to maintain this list here.

Reply:
Yeah I'll just add it to the metadata. This needs to be a function outside of the auth providers themselves, in order to safely ensure that the tables get dropped.

Reply:
Fixed